src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPOpenIdAuthenticator.java

```diff
@@ -0,0 +1,289 @@
+     * 1. Validate the OP is correct via TLS certificate check
+     * 2. Validate response is OK
+     * 3. Validate application type is either JSON or JWT
+     * 4. Validate response is one of: just signed, just encrypted, or signed then encrypted (encryption MUST NOT occur before signing)
+     * 5. If response is signed then validate the signature via JWS and confirm the response has "iss" and "aud" claims
+     * 6. Validate "iss" claim is equal to the issuer ID url
+     * 7. Validate the "aud" claim is equal to the client ID
+     * 8. If the client provides a userinfo_encrypted_response_alg value decrypt the response using the keys from registration
+     * 9. Validate "sub" claim is always present
+     * 10. Validate "sub" claim matches the ID token
+     * @param request The SecurityRequest to perform auth on
+     * @param context The active thread context
+     * @return AuthCredentials formed through querying the userinfo_endpoint
+     * @throws OpenSearchSecurityException On failure to extract credentials from the request
+     */
+    public AuthCredentials extractCredentials0(SecurityRequest request, ThreadContext context) throws OpenSearchSecurityException {
+
+        try {
+
+            URI userInfoEndpointURI = new URI(this.userInfoEndpoint);
+
+            String bearerHeader = request.getHeaders().get(AUTHORIZATION).getFirst();
+            if (!StringUtils.isBlank(bearerHeader)) {
+                if (bearerHeader.contains("Bearer ")) {
+                    bearerHeader = bearerHeader.substring(7);
+                }
+            }
+
+            String finalBearerHeader = bearerHeader;
+
+            AccessToken accessToken = new AccessToken(AccessTokenType.BEARER, finalBearerHeader) {
+                @Override
+                public String toAuthorizationHeader() {
+                    return "Bearer " + finalBearerHeader;
+                }
+            };
+
+            UserInfoRequest userInfoRequest = new UserInfoRequest(userInfoEndpointURI, accessToken);
+
+            HTTPRequest httpRequest = userInfoRequest.toHTTPRequest();
+
+            HTTPResponse httpResponse = httpRequest.send();
+            if (httpResponse.getStatusCode() < 200 || httpResponse.getStatusCode() >= 300) {
+                throw new AuthenticatorUnavailableException(
+                    "Error while getting " + this.userInfoEndpoint + ": " + httpResponse.getStatusMessage()
+                );
+            }
+
+            try {
+
+                UserInfoResponse userInfoResponse = UserInfoResponse.parse(httpResponse);
+
+                if (!userInfoResponse.indicatesSuccess()) {
+                    throw new AuthenticatorUnavailableException(
+                        "Error while getting " + this.userInfoEndpoint + ": " + userInfoResponse.toErrorResponse()
+                    );
+                }
+
+                String contentType = String.valueOf(httpResponse.getHeaderValues("content-type"));
```

Comment:
Extract out the UserInfoSuccessResponse here and then call on UserInfoSuccessResponse.getEntityContentType to get the content type from the request.

Reply:
Will do